library/manageiq.py

```diff
@@ -64,14 +61,110 @@
 '''
 
 
-def update_required(client, module, providers_url, provider_id, hostname, port):
-    try:
-        result = client.get(providers_url + '/%d/?attributes=authentications,endpoints' % provider_id)
+class ManageIQ(object):
+    """ ManageIQ object to execute various operations in manageiq
+
+    url      - manageiq environment url
+    user     - the username in manageiq
+    password - the user password in manageiq
+    """
+    openshift_provider_type = 'ManageIQ::Providers::OpenshiftEnterprise::ContainerManager'
```

Comment:
[for later PR] Ideally this would be module param e.g.
```
  manageiq_provider:
    type: openshift
    name: My Cluster
    ...
```
Should be required, even if currently we support only one (or two) type — there's no sensible default.